### PR TITLE
Unlock all languages

### LINKED ISSUE #29: Support for extra out-of-region languages
Netflix has added support for extra out-of-region subtitle languages. These subs are present in the manifest, but they are missing their download url. For in-region subtitles we get the download url from `manifest.timedtexttracks[].ttDownloadables`, but this field is empty for the extra subs. I have done only limited investigation at the time of writing, but I don't see the download url anywhere in other fields.

What's interesting is that once you select one of the extra languages as primary sub, the `ttDownloadables` for that track is populated with a download url - making that language available as secondary sub (and it will show up on refresh). This may be a deliberate obfuscation. It also indicates that `ttDownloadables` is still used by the player when activating the extra subs, we just need to figure out where it gets populated from. If we cannot easily get a complete list of download urls, we could potentially mimic user behaviour and just cycle through the primary subs one by one to intercept all urls (not ideal but 🤷‍♀️). 

I'm opening this issue to document the implementation of support for out of region languages, and request help from the community. Feedback, suggestions and PR are very much appreciated.

### PR DESCRIPTION
**Pull request by hyww**
resolves #29

Let me try again. This time I've tested with different versions of cadium-playercore [6.0033.414.911](https://github.com/truedread/netflix-1080p), [6.0034.588.911](https://github.com/TheGoddessInari/netflix-1080p-firefox), 6.0038.181.911 (latest?)

### CODE REVIEW

**Review by gmertes — COMMENTED** [review]
Still unable to reproduce :( I tried all versions of the player (original on Chrome and FF, truedread/netflix-1080p on Chrome and TheGoddessInari/netflix-1080p-firefox on FF). Setting `supportsPartialHydration` to false even disables all the extra languages for me (and I only see the 'original' languages in the primary subtitle list).

Are you sure you can get it to reproduce even with the original player and no other extensions installed? Maybe it's a regional difference? Does it reproduce for you on all shows?

**Comment by hyww** [comment]
After some experiments watching [Wednesday Ep.1](https://www.netflix.com/watch/81257204?trackId=200257859) with VPN to different countries, I believe it is a regional difference.

With VPN to UK, I got only these 'original' languages. (I prefer call them 'recommended' languages.) I guess this is similar with what you've seen (except `zh-Hant`, which is my Netflix profile preference).

```js
> __NflxMultiSubs.manifestManager.manifests[81257204].timedtexttracks.filter(a=>a.rank!=-1).map(a=>a.language)
['zh-Hant', 'en', 'ar', 'fr', 'pl', 'uk', 'fr', 'fr']
```

From Taiwan or with VPN to US, Japan, I got these

```js
['en', 'zh-Hant', 'ar', 'ca', 'cs', 'da', 'de', 'el', 'es-ES', 'es', 'eu', 'fil', 'fi', 'fr', 'gl', 'he', 'hr', 'hu', 'id', 'it', 'ja', 'ko', 'ms', 'nb', 'nl', 'pl', 'pt-BR', 'pt', 'ro', 'ru', 'sv', 'th', 'tr', 'uk', 'vi', 'zh-Hans', 'cs', 'es', 'fr', 'hu', 'pt-BR', 'hu', 'fr', 'cs']
```

My hypothesis is that Netflix is deprecating `showAllSubDubTracks`, but this change takes time to apply to all regions. Thus this PR is a work-around for some regions for now, not a long-term solution 😥

**Comment by gmertes** [comment]
I'm reluctant to merge even if it does work now for some regions, because it makes things worse for other regions by hiding the 'extra' languages from the primary subs list. Does it work for you without `supportsPartialHydration = false` ?

Many thanks for your work in investigating this! Would be great if we can get this to work for everyone.
